truss/templates/trtllm/packages/build_engine_utils.py

```diff
@@ -0,0 +1,129 @@
+from enum import Enum
+from pathlib import Path
+from typing import Optional
+
+from pydantic import BaseModel, ConfigDict
+
+# todo(Abu): consolidate this with build_engine.py
+# Enums / BaseClass is duplicated from build_engine so that we know what args / configurations
+# are available to us at run-time.
+
+
+class Quant(Enum):
+    NO_QUANT = "no_quant"
+    WEIGHTS_ONLY = "weights_only"
+    WEIGHTS_KV_INT8 = "weights_kv_int8"
+    SMOOTH_QUANT = "smooth_quant"
+
+
+class EngineType(Enum):
+    LLAMA = "llama"
+    MISTRAL = "mistral"
+
+
+class ArgsConfig(BaseModel):
+    max_input_len: Optional[int] = None
+    max_output_len: Optional[int] = None
+    max_batch_size: Optional[int] = None
+    tp_size: Optional[int] = None
+    pp_size: Optional[int] = None
+    world_size: Optional[int] = None
+    gather_all_token_logits: Optional[bool] = None
+    multi_block_mode: Optional[bool] = None
+    remove_input_padding: Optional[bool] = None
+    use_gpt_attention_plugin: Optional[str] = None
+    paged_kv_cache: Optional[bool] = None
+    use_inflight_batching: Optional[bool] = None
+    enable_context_fmha: Optional[bool] = None
+    use_gemm_plugin: Optional[str] = None
+    use_weight_only: Optional[bool] = None
+    output_dir: Optional[str] = None
+    model_dir: Optional[str] = None
+    ft_model_dir: Optional[str] = None
+    dtype: Optional[str] = None
+    int8_kv_cache: Optional[bool] = None
+    use_smooth_quant: Optional[bool] = None
+    per_token: Optional[bool] = None
+    per_channel: Optional[bool] = None
+    parallel_build: Optional[bool] = None
+
+    # to disable warning because `model_dir` starts with `model_` prefix
+    model_config = ConfigDict(protected_namespaces=())
+
+    def as_command_arguments(self):
+        non_bool_args = [
+            element
+            for arg, value in self.dict().items()
+            for element in [f"--{arg}", str(value)]
```

Comment:
Nit: consider `=`-style flags. I personally find it easier to visually parse into arg_name arg_value pairs than spaces...